### PR TITLE
radix__sort.py

### PR DESCRIPTION
**Pull request by abhishekfrayne**
### Describe your change:

* [x] Add an algorithm? Yes
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: sorts/radix__sort.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Python program for implementation of Radix Sort
+
+# A function to do counting sort of arr[] according to
+# the digit represented by exp.
+def counting_sort(arr, exp1):
```
Comment (abhishekfrayne) [inline sorts/radix__sort.py:5]: def defining counting_sort

File: sorts/radix__sort.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    # The output array elements that will have sorted arr
+    output = [0] * (n)
+
+    # initialize count array as 0
+    count = [0] * (10)
+
+    # Store count of occurrences in count[]
+    for i in range(0, n):
+        index = arr[i] / exp1
```
Comment (abhishekfrayne) [inline sorts/radix__sort.py:16]: Initialising variables and comments to explain code.

File: sorts/radix__sort.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    while i >= 0:
+        index = arr[i] / exp1
+        output[count[int((index) % 10)] - 1] = arr[i]
+        count[int((index) % 10)] -= 1
+        i -= 1
+
+    # Copying the output array to arr[],
+    # so that arr now contains sorted numbers
+    i = 0
+    for i in range(0, len(arr)):
```
Comment (abhishekfrayne) [inline sorts/radix__sort.py:35]: Initialising for loop to be ran

File: sorts/radix__sort.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+		i -= 1
+
+	# Copying the output array to arr[], 
+	# so that arr now contains sorted numbers 
+	i = 0
+	for i in range(0,len(arr)): 
+		arr[i] = output[i] 
+
+# Method to do Radix Sort
+def radix_sort(arr):
```
[the remark this answers is not in the document]
Comment (abhishekfrayne) [reply sorts/radix__sort.py:40]: Initialising variables and comments to explain code.

**Comment by cclauss** [comment]
Closing require_type_hints PRs to prepare for Hacktoberfest

**Pull request closed without merging**